src/util/execute-lifecycle-script.js

```diff
@@ -202,28 +206,15 @@ export async function executeLifecycleScript(
     cmd = fixCmdWinSlashes(cmd);
   }
 
-  let updateProgress;
-  if (spinner) {
-    updateProgress = data => {
-      const dataStr = data
-        .toString() // turn buffer into string
-        .trim(); // trim whitespace
-
-      invariant(spinner && spinner.tick, 'We should have spinner and its ticker here');
-      if (dataStr) {
-        spinner.tick(
-          dataStr
-            // Only get the last line
-            .substr(dataStr.lastIndexOf('\n') + 1)
-            // change tabs to spaces as they can interfere with the console
-            .replace(/\t/g, ' '),
-        );
-      }
-    };
-  }
+  // if we aren't interactive then pipe everything to the terminal
+  const stdio = isInteractive ? 'inherit' : undefined;
+  // if this is not interactive, run child processes detached so they can't hang
+  // trying to read from /dev/tty but not on Windows, because Windows opens
+  // a new console window and there is no /dev/tty anyway
+  const detached = !isInteractive && process.platform !== 'win32';
```

Comment:
I find it slightly more readable this way, wdyt?
```
// By default (non-interactive), pipe everything to the terminal and run child process detached
// as long as it's not Windows (since windows does not have /dev/tty)
let stdio;
let detached = process.platform !== 'win32';

if (isInteractive) {
   stdio = 'inherit';
   detached = false;
}
```

Reply:
Don't really like using `let` there but I also agree that your version expresses the intent more clearly. I'll see if I can find a middle ground :)

Reply:
I ended up using your version since it is clearly more readable.